tensorflow/c/experimental/filesystem/modular_filesystem.cc

```diff
@@ -390,6 +390,39 @@ void ModularFileSystem::FlushCaches(TransactionToken* token) {
   if (ops_->flush_caches != nullptr) ops_->flush_caches(filesystem_.get());
 }
 
+Status ModularFileSystem::SetConfiguration(const std::string& name,
+                                           const std::vector<string>& values) {
+  if (ops_->set_filesystem_configuration == nullptr) {
+    return errors::Unimplemented(
+        "Filesystem does not support SetConfiguration()");
+  }
+  if (values.size() == 0) {
+    return errors::InvalidArgument(
+        "SetConfiguration() needs number of values > 0");
+  }
+
+  TF_Filesystem_Option option;
+  memset(&option, 0, sizeof(option));
+  option.name = const_cast<char*>(name.c_str());
+  option.per_file = 0;
+  TF_Filesystem_Option_Value option_value;
+  memset(&option_value, 0, sizeof(option_value));
+  option_value.type_tag = TF_Filesystem_Option_Type_Buffer;
+  option_value.num_values = values.size();
+  std::vector<TF_Filesystem_Option_Value_Union> option_values(values.size());
```

Comment:
Why allocate a new `std::vector`? We can use `new TF_Filesystem_Option_Value_Union[values.size()]`;